src/include/platform/PlatformManager.h

```diff
@@ -183,24 +201,18 @@ class PlatformManager
      * returns.
      */
     CHIP_ERROR StopEventLoopTask();
-    void LockChipStack();
-    bool TryLockChipStack();
-    void UnlockChipStack();
-    void Shutdown();
 
-#if CHIP_STACK_LOCK_TRACKING_ENABLED
-    bool IsChipStackLockedByCurrentThread() const;
+    // These functions should be gated by CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+    // but the PlatformManagerImpl header (included below) makes that tricky
+    // so always declare them here even if the are not defined by the platform implementation.
+#if 1 || defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+    CHIP_ERROR ScheduleBackgroundWork(AsyncWorkFunct workFunct, intptr_t arg = 0);
```

Comment:
These should be documented, even if just to say that they have the same semantics as the non-"Background" functions.

Reply:
I've added some comments to that effect. Let me know if more needs to be said.